portfolio/src/main/webapp/script.js

```diff
@@ -89,6 +94,29 @@ function makeComment(heading, content) {
     return row;
 }
 
+function addDeleteButton(heading, commentId) {
+    const th = document.createElement("th");
+    th.innerText = heading;
+
+    const form = document.createElement("form");
+    form.setAttribute("action", "/delete-data");
+    form.setAttribute("method", "POST");
+
+    const button = document.createElement("button");
+    button.setAttribute("type", "submit");
+    button.innerText = "delete comment";
+
+    const id = document.createElement("input");
+    id.name = "comment-id"
+    id.setAttribute("type", "hidden");
+    id.value = commentId;
+
+    form.appendChild(button);
+    form.appendChild(id);
+    th.append(form);
+    return th;
+}
```

Comment:
Creating too much HTML in JS makes it difficult to debug or see where certain things on the page are coming from. Rather than creating a full form here, I would recommend either just making a button with a callback method, or having the form in a template and bringing in that template html and modifying the template as necessary. Or maybe the button/form can always be there but just marked `display=none;` when it shouldn't be present or something.

Reply:
fixed!